### PR TITLE
Order files before processing

### LINKED ISSUE #599: Different ordering of file processing on different platforms/systems
## Version
HEAD

## 🐞 Description
When a folder path is passed to Pepys, it processes all files in that folder. It seems that on different systems, it can process the files in different orders - which can lead to some tests failing if they depend on the order.

## ⚙️ Current behavior
The order in which files are processed differs between OS X and Linux, and possibly also between Travis Linux and Github Actions Linux. This causes problems when, for example, the first entry of a list is checked - but because the files were processed in a different order, it is actually the 3rd entry (for example) which is the one that should be checked.

## ⚙️ Expected behavior
Files should be processed in the same order regardless of platform and operating system, or tests should be designed which are not dependent on the ordering of files. Possibly both approaches should be taken.

## 🔢 Steps to Reproduce
It's easiest just to look at the [test results on Github Actions](https://github.com/debrief/pepys-import/actions/runs/347925850) which show three test failures. I've resolved these by getting each of these tests to just process a single file rather than a folder - which is probably a good thing to do anyway, as they don't need to process multiple files, and this speeds up the tests anyway.

## Discussion
I'm open to input as to whether we should do something with the file processing code to sort the file lists in Python before going through them, as the Python sorting functions should be platform-independent.

### PR DESCRIPTION
**Pull request by BarisSari**
## 🧰 Issue
Fixes #599 

## 🤔 Reason: 

`os.listdir()` and `os.scandir()` returns files in arbitrary order. So, the order of the processing files change and they depend on the operating systems. This PR adds ordering to get repeatable results.

## 🔨Work carried out:

- [x] Sorting added to each line that calls `os.listdir` or `os.scandir` methods.
- [x] Tests pass

## Confirmations

- [ ] I have chosen reviewers for my PR.
- [x] I have chosen an appropriate label for the PR.
- [x] I have completed the mandatory sections of this document.
- [x] I have deleted any unused sections.

## 📝 Developer Notes:
It might be better to review & merge this PR after Robin's Github Action work #598  is done.

### CODE REVIEW

**Review by IanMayo — APPROVED** [review]

**Comment by IanMayo** [comment]
@BarisSari  - did you want to defer this PR until after #598 so that it can be used to test #598 ?

It doesn't seem to have any relevance/dependency on that PR.

File: pepys_import/utils/import_utils.py [hunk 1]
```diff
@@ -31,11 +31,15 @@ def import_validators(path):
...
             if file.is_file():
                 classes = import_module_(file)
                 for _, class_ in classes:
                     print(inspect.signature(class_))
                     validators.append(class_)
     return validators
+
+
+def sort_files(files):
+    return sorted(files, key=lambda x: x.name)
```
[the remark this answers is not in the document]
Comment (BarisSari) [reply pepys_import/utils/import_utils.py:45]: It's used for one line functions, @IanMayo. It's equal to the following:

```python
def function(x):
    return x.name
```

**Comment by BarisSari** [comment]
> @BarisSari - did you want to defer this PR until after #598 so that it can be used to test #598 ?
> 
> It doesn't seem to have any relevance/dependency on that PR.

Yes, Ian. Linux was always running correctly but it would be better to test it on Windows, too. That's why I wanted to defer it.

**Comment by BarisSari** [comment]
Tests passed for Windows as well. I think this PR is ready to merge, @IanMayo.

**Pull request merged**